### ISSUE TITLE
MNT: Remove is_path_hidden and walk_skip_hidden

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#14799]
When: Not before v7 (as it was deprecated in v6) but actual timeline might depend on:

* https://github.com/astropy/astropy-APEs/pull/82

Remove stuff deprecated in https://github.com/astropy/astropy/pull/14759 , `test_is_path_hidden_deprecation`, `test_skip_hidden`, `astropy.utils.tests = data/.hidden_file.txt` in package data.

**Comment by marcellonascif** [comment]
Heyy, I know this issue doesn`t need to be done before v7 but I`d like to work on this now and leave a PR so you can use it on v7. May I work on this?
thanks!

**Comment by pllim** [comment]
No, still a year away. We have not branched for v6.0, let alone v6.1, before v7. Please choose another issue to work on. Thanks for your interest!

**Comment by NathanMuniz18** [comment]
hello guys, I would like to help with this issue and would like to know if I have permission to contribute to it?

**Comment by pllim** [comment]
Might have to wait till v8, I think, so not yet? Please find another issue for now. Thanks!

**Comment by anaclaragalvao** [comment]
hey guys, could I work on this issue?

**Comment by neutrinoceros** [comment]
Yes. It should be pretty straightforward. Thank you
